conftest.py

```diff
@@ -85,6 +89,32 @@ def run(self, **kwargs: typing.Any):
         )
         return self.container
 
+    def run_and_wait(
+        self,
+        timeout: int,
+        no_warnings: bool = True,
+        no_errors: bool = True,
+        **kwargs: typing.Any,
+    ) -> str:
+        running_container = self.run_detached(**kwargs)
+        rv = running_container.wait(timeout=timeout)
+        logs = running_container.logs().decode("utf-8")
+        LOGGER.debug(logs)
+        if no_warnings:
+            assert not self.get_warnings(logs)
+        if no_errors:
+            assert not self.get_errors(logs)
+        assert rv == 0 or rv["StatusCode"] == 0
+        return logs
+
+    @staticmethod
+    def get_warnings(logs: str) -> list[str]:
+        return [line for line in logs.split("\n") if line.startswith("WARNING")]
+
+    @staticmethod
+    def get_errors(logs: str) -> list[str]:
+        return [line for line in logs.split("\n") if line.startswith("ERROR")]
```

Comment:
This is partially not the same as it was, but I think we should only worry about our errors and not the whole log output.

Reply:
I would make a small change to avoid code duplication.

```python
@staticmethod
def get_errors(logs: str) -> list[str]:
    return _lines_starting_with(logs, "ERROR")

@staticmethod
def get_warnings(logs: str) -> list[str]:
    return _lines_starting_with(logs, "WARNINGS")

@staticmethod
def _lines_starting_with(logs: str, pattern: str) -> list[str]:
    return [line for line in logs.splitlines() if line.startswith(pattern)]
```

Reply:
Done as you suggested (but it should be `WARNING`, not `WARNINGS`).

Reply:
OK.